pkg/services/ngalert/notifier/redis_peer.go

```diff
@@ -90,13 +94,25 @@ func newRedisPeer(cfg redisConfig, logger log.Logger, reg prometheus.Registerer,
 	if cfg.maxConns >= 0 {
 		poolSize = cfg.maxConns
 	}
-	rdb := redis.NewClient(&redis.Options{
+
+	opts := &redis.Options{
 		Addr:     cfg.addr,
 		Username: cfg.username,
 		Password: cfg.password,
 		DB:       cfg.db,
 		PoolSize: poolSize,
-	})
+	}
+
+	if cfg.tlsEnabled {
+		tlsClientConfig, err := cfg.tls.GetTLSConfig()
+		if err != nil {
+			logger.Error("Failed to get TLS config", "err", err)
```

Comment:
I would return the error here, a quick skim at the errors from `GetTLSConfig` shows they are mostly configuration issues and are likely unrecoverable. I believe the argument for not failing on a failed Ping lower down the code, is because the failure is transient and could resolve after startup. 

```suggestion
			logger.Error("Failed to get TLS config", "err", err)
			return nil, err
```